aoc/cmd/root.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"aoc/pkg/advent"
+	"aoc/pkg/year_2022"
+	"github.com/spf13/cobra"
+	"log"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "my-aoc",
+	Short: "Compute and submit Advent Of Code solutions",
+	Long: "my-aoc lets you submit the solution to any day of Advent Of Code by running your text input against the solutions provided in this project. Please" +
+		" provide the AOC_SESSION_ID environment variable with your session ID when interacting with this binary.",
+	Run: Run,
+}
+
+var Submit bool = false
+
+func init() {
+	rootCmd.PersistentFlags().IntP("day", "d", 1, "The day to run solutions for.")
+	rootCmd.PersistentFlags().IntP("year", "y", 2022, "The year to run solutions for. Currently, only 2022 is supported.")
+	rootCmd.PersistentFlags().IntP("part", "p", 1, "The part to run solution for.")
+	rootCmd.PersistentFlags().BoolVarP(&Submit, "submit", "s", false, "If set, will submit your solution to AOC.")
+
+	_ = rootCmd.MarkFlagRequired("verbose")
+	_ = rootCmd.MarkFlagRequired("day")
+	_ = rootCmd.MarkFlagRequired("year")
+	_ = rootCmd.MarkFlagRequired("part")
+}
+
+func Run(cmd *cobra.Command, args []string) {
+	flags := cmd.Flags()
+	day, _ := flags.GetInt("day")
+	year, _ := flags.GetInt("year")
+	part, _ := flags.GetInt("part")
+
+	if year != 2022 {
+		log.Fatal("Currently only have solutions for 2022. Please use 2022 as year flag or skip that flag altogether.")
+	}
+
+	inputs, err := advent.GetInputs(day, year)
+	if err != nil {
+		log.Fatalf("Could not fetch inputs: %s", err.Error())
+	}
+
+	switch day {
```

Comment:
You can create array/map of `Solution` type with initialized pointers to each solutions implementation to avoid switch with repetitive code.